ext/src/ruby_api/store.rs

```diff
@@ -115,19 +117,25 @@ impl Store {
     ///   store = Wasmtime::Store.new(Wasmtime::Engine.new, {})
     pub fn new(ruby: &Ruby, args: &[Value]) -> Result<Self, Error> {
         let args = scan_args::scan_args::<(&Engine,), (Option<Value>,), (), (), _, ()>(args)?;
-        let kw = scan_args::get_kwargs::<_, (), (Option<&WasiCtxBuilder>,), ()>(
-            args.keywords,
-            &[],
-            &[*WASI_CTX],
-        )?;
         let (engine,) = args.required;
         let (user_data,) = args.optional;
         let user_data = user_data.unwrap_or_else(|| ().into_value());
-        let wasi = match kw.optional.0 {
-            None => None,
-            Some(wasi_ctx_builder) => Some(wasi_ctx_builder.build_context(ruby)?),
+
+      
+        let kw = scan_args::get_kwargs::<_, (), (Option<&WasiCtxBuilder>, Option<&WasiDeterministicCtxBuilder>), ()>(
+            args.keywords,
+            &[],
+            &[*WASI_CTX, *WASI_DET_CTX],
```

Comment:
There is a better way to do this, instead of creating addition kwargs. But I'd like to confirm it's working in flow-runtime and then will focus on refactoring.